Disable flaky tests.

Rosalita opened the pull request:
The following tests have been seen to flake in CI, this PR disables them.

- Test_RTN4a_ConnectionEventForStateChange
- TestAuth_BasicAuth
- TestAuth_ClientID

https://github.com/ably/ably-go/issues/438 has been created to
investigate fixing them and turning them back on

Rosalita commented:
Some background to this PR. 

Recently a PR was raised to this code base, out of 4 test runs for that PR the following was observed.

Attempt #1 all tests failed to start - this run was ignored.

Attempt #2 the follow test failed on Go 1.16

```
=== RUN   Test_RTN4a_ConnectionEventForStateChange
=== RUN   Test_RTN4a_ConnectionEventForStateChange/on_CONNECTING
    ably_test.go:66: safeclose 1: failed to close *ablytest.Sandbox: Delete "https://sandbox-rest.ably.io/apps/_tmp_aRmoog": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
=== RUN   Test_RTN4a_ConnectionEventForStateChange/on_CONNECTED
    realtime_conn_spec_test.go:148: timed out waiting for channel receive
    ably_test.go:66: safeclose 0: failed to close ablytest.realtimeIOCloser: context deadline exceeded
=== RUN   Test_RTN4a_ConnectionEventForStateChange/on_DISCONNECTED
--- FAIL: Test_RTN4a_ConnectionEventForStateChange (182.20s)
    --- PASS: Test_RTN4a_ConnectionEventForStateChange/on_CONNECTING (61.86s)
    --- FAIL: Test_RTN4a_ConnectionEventForStateChange/on_CONNECTED (68.89s)
    --- FAIL: Test_RTN4a_ConnectionEventForStateChange/on_DISCONNECTED (51.44s)
panic: request error: Service Unavailable ("{\"message\":\"Ably's routing layer was unable to service this request\",\"code\":50305,\"statusCode\":503,\"href\":\"https://help.ably.io/error/50000\"}") [recovered]
	panic: request error: Service Unavailable ("{\"message\":\"Ably's routing layer was unable to service this request\",\"code\":50305,\"statusCode\":503,\"href\":\"https://help.ably.io/error/50000\"}")
```

And the following test failed on Go 1.17

```
=== RUN   TestAuth_BasicAuth
--- FAIL: TestAuth_BasicAuth (0.53s)
panic: request error: Service Unavailable ("{\"message\":\"Ably's routing layer was unable to service this request\",\"code\":50305,\"statusCode\":503,\"href\":\"https://help.ably.io/error/50000\"}") [recovered]
	panic: request error: Service Unavailable ("{\"message\":\"Ably's routing layer was unable to service this request\",\"code\":50305,\"statusCode\":503,\"href\":\"https://help.ably.io/error/50000\"}")
```


Attempt #3 The following tests failed on Go 1.16

```
=== RUN   TestAuth_ClientID
    auth_test.go:673: Close()=waiting on Result timed out after 30s
--- FAIL: TestAuth_ClientID (37.98s)
```

Attempt #4 all tests passed.

These tests are failing due to timeout and a service which was unavailable.

QuintinWillison reviewed (approved):
Ideally it would be nice to see more context at the point you're disabling these, but equally you could add that to the issue you created. An example of where this context has been added for another skipped test is here:

https://github.com/ably/ably-go/blob/6cf9e4e060a67444ec02d587ae0e93c24ba96d7f/ably/realtime_client_test.go#L178

Rosalita commented:
> Ideally it would be nice to see more context at the point you're disabling these, but equally you could add that to the issue you created. An example of where this context has been added for another skipped test is here:
> 
> https://github.com/ably/ably-go/blob/6cf9e4e060a67444ec02d587ae0e93c24ba96d7f/ably/realtime_client_test.go#L178

I've added the context to the Github issue and added a comment in code next to each skip with a link to the Github issue.

Rosalita commented:
I don't have sufficient permission to merge this PR so @lmars or @QuintinWillison will need to merge if they are happy for this to land.
